Remove unneeded array and handle bigendian

Flova opened the pull request:
This pull request replaces the `pftype_sizes` mapping with a query to `type_mappings` using the `itemsize` attribute of the returned dtype.

It also adds support for big-endian CPU architectures.

Flova committed:
Cleanup

Flova commented:
Still needs bigendian support for reading point clouds.

Flova commented:
Friendly ping @tpanzarella

tpanzarella commented on ros2_numpy/point_cloud2.py:
```diff
@@ -159,7 +152,7 @@ def array_to_pointcloud2(cloud_arr, stamp=None, frame_id=None):
     cloud_msg.width = cloud_arr.shape[1]
     cloud_msg.fields = dtype_to_fields(cloud_arr.dtype)
-    cloud_msg.is_bigendian = False # assumption
+    cloud_msg.is_bigendian = sys.byteorder != 'little',
```
The trailing `,` here results in setting `cloud_msg.is_bigendian` to a `tuple` and not a `bool` as you intend.

Flova replied:
Oh this was a leftover from setting it in the constructor.

tpanzarella reviewed (changes requested):
Both parts of this PR are failing the unit tests. See output of the following for details:

```
colcon test --event-handlers console_cohesion+ && colcon test-result --verbose
```

As noted above, there are two parts to this PR and they are unrelated. In general, it would be better to separate unrelated concerns like this into two separate PRs.

Flova committed:
Remove ,

Flova committed:
USe correct mapping field

Flova committed:
Fix for > Python 3.10

Flova commented:
I fixed all of the above issues. I also noticed some error while running the tests under rolling.
`collections` is no longer exposing `Sequence`, so it needs to be imported via the previously recommended way by using `collections.abc`. I fixed this in this PR to make the tests pass in rolling.

Flova commented:
This PR is manily based on the discussion in https://github.com/ros2/common_interfaces/pull/175, largely https://github.com/ros2/common_interfaces/pull/175#discussion_r823259720. This is also the reason why this is one PR, combining the things I noticed while doing the other PR. As the PR in https://github.com/ros2/common_interfaces is merged now we could also go the route of wrapping the official numpy pointcloud API for `ros2_numpy`. This should unify implementations and reduce maintenance overhead.

Flova commented:
Friendly ping @tpanzarella
